File: 448-FindAllNumbersDisappearedInAnArray/448-FindAllNumbersDisappearedInAnArray.java
// Last updated: 5/25/2026, 11:09:32 AM
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        List<Integer>a=new ArrayList<>();
        int index=0;
        while(index<nums.length){
            int z=nums[index]-1;
            if(nums[index]!=nums[z]){
                int t=nums[index];
                nums[index]=nums[z];
                nums[z]=t;
            }
            else{
                index++;
            }
        }
        for(int i=0;i<nums.length;i++){
            if(nums[i]!=i+1) a.add(i+1);
        }
        return a;
    }
}

File: 448-FindAllNumbersDisappearedInAnArray/448-FindAllNumbersDisappearedInAnArray.py
# Last updated: 5/25/2026, 11:09:38 AM
class Solution(object):
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        # Step 1: Mark each visited index as negative
        for num in nums:
            index = abs(num) - 1
            nums[index] = -abs(nums[index])

        # Step 2: Collect indices which are still positive
        result = []
        for i in range(len(nums)):
            if nums[i] > 0:
                result.append(i + 1)

        return result
        